### DISCUSSION TITLE
Do we need to specify the `@babel/plugin-transform-runtime`'s `version` option?

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#14444] [Q&A] [answered]
As title. From [docs](https://babeljs.io/docs/en/babel-plugin-transform-runtime#version):

> By default transform-runtime assumes that @babel/runtime@7.0.0 is installed. If you have later versions of @babel/runtime (or their corejs counterparts e.g. @babel/runtime-corejs3) installed or listed as a dependency, transform-runtime can use more advanced features.

It's still not very clear that:

1. The default is `7.0.0`?
2. If I have a higher version `@babel/runtime` installed, it's still `7.0.0`? or it will automatically sync. up to the latest one I can use?

**Comment by JLHwung** [comment, marked as answer]
> The default is 7.0.0?

Yes, otherwise upgrading `@babel/plugin-transform-runtime` will break things when users have an old version of `@babel/runtime`.

> If I have a higher version @babel/runtime installed, it's still 7.0.0? or it will automatically sync. up to the latest one I can use?

In Babel 8 we will warn on the version discrepancy: https://github.com/babel/babel/pull/10325. But detecting the version of `@babel/runtime` used by end users may be still too far-fetched. That said, if you are bundling an app and you have control of the `@babel/runtime` version, you can always explicitly let `transform-runtime` know:

```js
{
  version: require("@babel/runtime/package.json").version
}
```

**Reply by nyngwang** [reply]
Very clear, thanks!
